[BUG] Todo check fails to decode Unicode character

### Version / Commit

_No response_

### What happened?

When executing format, the following is logged:

```sh
/cmake --build /nebulastream-public/cmake-build-debug --target format -j 10
[0/1] cd /nebulastream-public && scripts/format.sh -i
>checking added TODOs for last 8 commits (i.e. since forking off 'main')
Set env var BASE_REF to override base, e.g. call:

  BASE_REF=origin/main python3 scripts/check_todos.py
Traceback (most recent call last):
  File "/nebulastream-public/scripts/check_todos.py", line 136, in <module>
    main()
  File "/nebulastream-public/scripts/check_todos.py", line 46, in main
    diff = run_cmd(["git", "diff", "--merge-base", base])
  File "/nebulastream-public/scripts/check_todos.py", line 16, in run_cmd
    p = subprocess.run(cmd, capture_output=True, check=True, text=True)
  File "/usr/lib/python3.10/subprocess.py", line 505, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/usr/lib/python3.10/subprocess.py", line 1154, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/usr/lib/python3.10/subprocess.py", line 2059, in _communicate
    stdout = self._translate_newlines(stdout,
  File "/usr/lib/python3.10/subprocess.py", line 1031, in _translate_newlines
    data = data.decode(encoding, errors)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xaf in position 123101: invalid start byte
FAILED: CMakeFiles/format /nebulastream-public/cmake-build-debug/CMakeFiles/format 
cd /nebulastream-public && scripts/format.sh -i
ninja: build stopped: subcommand failed.
```

### What is the expected behavior?

The script should generally not fail, it should just notify about open todos that need to be addressed.

### How to reproduce

Problem occurs on the latest version of the branch corresponding to #74.

### Operating System

_No response_

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/nebulastream/nebulastream/blob/main/CODE_OF_CONDUCT.md)

--- a/scripts/check_todos.py
+++ b/scripts/check_todos.py
@@ -13,8 +13,8 @@ import urllib.request
 
 def run_cmd(cmd: list) -> str:
     try:
-        #ignoring errors to pass format check. Proper fix in: Todo #343
-        p = subprocess.run(cmd, capture_output=True, check=True, text=True, errors='ignore')
+        # Handle decode errors by replacing illegal chars with � (see #343)
+        p = subprocess.run(cmd, capture_output=True, check=True, text=True, errors='replace')
     except subprocess.CalledProcessError as e:
         print(e)
         print("\nstderr output:")

